core/meterer/meterer.go

```diff
@@ -214,14 +214,20 @@ func (m *Meterer) ServeOnDemandRequest(ctx context.Context, header core.PaymentM
 		return fmt.Errorf("invalid quorum for On-Demand Request: %w", err)
 	}
 
+	// Compute charge at payment time
```

Comment:
is it possible to move this block into the deleted `PaymentCharged` helper fn? Would like to keep those unit tests if possible?